src/main/java/org/datadog/jmxfetch/util/ByteArraySearcher.java

```diff
@@ -0,0 +1,47 @@
+package org.datadog.jmxfetch.util;
+
+public final class ByteArraySearcher {
+
+    private final long[] high;
+    private final long[] low;
+    private final long termination;
+
+    /**
+     * Constructs a simple literal matcher from the input term.
+     * @param term the input term, must be shorter than 64 bytes.
+     */
+    public ByteArraySearcher(byte[] term) {
+        if (term.length > 64) {
+            throw new IllegalArgumentException("term must be shorter than 64 characters");
+        }
+        this.high = new long[16];
+        this.low = new long[16];
+        int mask = 1;
+        for (byte b : term) {
+            low[b & 0xF] |= mask;
+            high[b >>> 4] |= mask;
+            mask <<= 1;
+        }
+        this.termination = 1 << (term.length - 1);
+    }
+
+
+    /**
+     * Returns true if the array contains a literal match of this searcher's term.
+     * @param array the input array
+     * @return whether the array matches or not
+     */
+    public boolean matches(byte[] array) {
+        long state = 0;
+        for (int i = 0; i < array.length; ++i) {
+            byte symbol = array[i];
```

Comment:
minor: this could use a foreach like the method above, since the value of `i` is only used to access the element

Reply:
I couldn't remember if this was legal in JDK6 since I was still writing C++ when it was popular, I suppose I should change this.